common/utils.js

```diff
@@ -1190,6 +1191,343 @@
+                    } else if (option.url) {
+                        option.type = "url";
+                    } else {
+                        isValid = false;
+                    }
+                    break;
+            }
+
+            return isValid;
+        }
+
+        // triggered when top level menu is opened/closed
+        function onToggle(open) {
+            var elems = document.querySelectorAll(".dropdown-menu.show");
+            [].forEach.call(elems, function(el) {
+                el.classList.remove("show");
+            });
+
+            // whenever a dropdown menu is closed, remove the child-opened class that adds highlight color
+            var highlightedParents = document.querySelectorAll(".dropdown-submenu.child-opened");
+            [].forEach.call(highlightedParents, function(el) {
+                el.classList.remove("child-opened");
+            });
+
+            // calculate height for each open dropdown menu
+            if (open) {
+                var openDropdowns = document.querySelectorAll(".dropdown.open ul");
+                [].forEach.call(openDropdowns, function(el) {
+                    checkHeight(el, window.innerHeight);
+                    _checkWidth(el, window.innerWidth);
+                });
+            }
+        }
+
+        /**
+         * Just to make sure browsers are not ignoring the ng-click, we are first
+         * preventing the default behavior of link, then logging the client action
+         * and then changing the location without waiting for the request,
+         * This will ensure that we're at least sending the log to server.
+         */
+        function onLinkClick() {
+            return function ($event, menuObject) {
+                $event.preventDefault();
+                $event.stopPropagation();
+
+                // NOTE: if link goes to a chaise app, client logging is not necessary (we're using ppid, pcid instead)
+                if (!isChaise(menuObject.url, ConfigUtils.getContextJSON())) {
+                    // check if external or internal resource page
+                    var action = UriUtils.isSameOrigin(menuObject.url) ? logService.logActions.NAVBAR_MENU_INTERNAL : logService.logActions.NAVBAR_MENU_EXTERNAL;
+                    logService.logClientAction({
+                        action: logService.getActionString(action, "", ""),
+                        names: menuObject.names
+                    });
+                }
+
+                if (menuObject.newTab) {
+                    $window.open(menuObject.url, '_blank');
+                } else {
+                    $window.location = menuObject.url;
+                }
+            };
+        }
+
+        function renderName(option) {
+            // new syntax will always use nameMarkdownPattern
+            if (option.nameMarkdownPattern) {
+                return $sce.trustAsHtml(ERMrest.renderHandlebarsTemplate(option.nameMarkdownPattern, {inline: true}));
```

Comment:
You need to do `renderHandlebarsTemplate` and then `renderMarkdown` (since it's both pattern and markdown).